tests/Subscriber/GitRepoSubscriberTest.php

```diff
@@ -11,67 +11,341 @@
+    }
+
+    public function testAllowsToOverwriteConfiguredOrgAndRepo()
+    {
+        $command = new GitRepoCommand();
+        $commandTest = $this->runCommandTest(
+            $command,
+            ['--repo-adapter' => 'github_enterprise', '--org' => 'cordoval', '--repo' => 'gush-sandbox'],
+            ['repo_adapter' => 'github', 'repo_org' => 'gushphp', 'repo_name' => 'gush']
+        );
+
+        $display = $commandTest->getDisplay(true);
+
+        $this->assertNotContains('You did not set or provide an adapter-name', $display);
+
+        $this->assertNotNull($command->getAdapter());
+        $this->assertEquals('github_enterprise', $command->getAdapter()->getAdapterName());
+
+        $this->assertEquals('cordoval', $command->getAdapter()->getUsername());
+        $this->assertEquals('gush-sandbox', $command->getAdapter()->getRepository());
+
+        // This correct as the issue-org/repo is not set and so inherits
+        // from the adapter-org/repo config. The option only overwrites the adapter.
+        $this->assertEquals('gushphp', $command->getIssueTracker()->getUsername());
+        $this->assertEquals('gush', $command->getIssueTracker()->getRepository());
+    }
+
+    public function testAllowsToOverwriteConfiguredOrgAndRepoForIssue()
+    {
+        $command = new GitRepoCommand();
+        $commandTest = $this->runCommandTest(
+            $command,
+            ['--repo-adapter' => 'github_enterprise', '--issue-org' => 'cordoval', '--issue-project' => 'gush-sandbox'],
+            ['repo_adapter' => 'github', 'repo_org' => 'gushphp', 'repo_repo' => 'gush']
+        );
+
+        $display = $commandTest->getDisplay(true);
+
+        $this->assertNotContains('You did not set or provide an adapter-name', $display);
+
+        $this->assertNotNull($command->getAdapter());
+        $this->assertEquals('github_enterprise', $command->getAdapter()->getAdapterName());
+
+        $this->assertEquals('gushphp', $command->getAdapter()->getUsername());
+        $this->assertEquals('gush', $command->getAdapter()->getRepository());
+
+        $this->assertEquals('cordoval', $command->getIssueTracker()->getUsername());
+        $this->assertEquals('gush-sandbox', $command->getIssueTracker()->getRepository());
+    }
+
+    public function testThrowsErrorOnInvalidRepoAdapter()
+    {
+        $this->setExpectedException(
+            UserException::class,
+            'Adapter "noop-noop" (for repository-management) does not seem to be supported'
+        );
+
+        $command = new GitRepoCommand();
+        $this->runCommandTest(
+            $command,
+            ['--repo-adapter' => 'noop-noop'],
+            ['repo_adapter' => 'github']
+        );
+    }
+
+    public function testThrowsErrorOnInvalidIssueAdapter()
+    {
+        $this->setExpectedException(
+            UserException::class,
+            'Adapter "noop-noop" (for issue-tracking) does not seem to be supported'
```

Comment:
not supported rather than leaving it to chance, even though if it could be another thing is best to use precision of language